fitbenchmarking/parsing/parser_factory.py

```diff
@@ -0,0 +1,73 @@
+"""
+This file contains a factory implementation for the parsers.
+This is used to manage the imports and reduce effort in adding new parsers.
+"""
+
+from importlib import import_module
+from inspect import isclass, isabstract, getmembers
+
+from fitbenchmarking.parsing.base_parser import Parser
+
+
+class ParserFactory:
+    """
+    A factory for creating parsers.
+    This has the capability to select the correct parser, import it, and
+    generate an instance of it.
+    Parsers generated from this must be a subclass of base_parser.Parser
+    """
+
+    @staticmethod
+    def create_parser(filename):
+        """
+        Inspect the input and create a parser that matches the required file.
+
+        :param filename: The path to the file to be parsed
+        :type filename: string
+
+        :returns: Initialised parser for the problem
+        :rtype: fitbenchmarking.parsing.base_parser.Parser subclass
+        """
+
+        with open(filename, 'r') as f:
+            line = f.readline()
+
+        # Take the first section of text
+        module_name = ''
+        for l in line.strip('#').strip():
+            if not l.isalpha():
+                break
+            module_name += l
+
+        module_name = '{}_parser'.format(module_name.lower())
+        try:
+            module = import_module('.' + module_name, __package__)
+        except ImportError:
+            raise ValueError('Could not find parser for {}.'.format(filename)
+                             + 'Check the input is correct and try again.')
+
+        classes = getmembers(module, lambda m: (isclass(m)
+                                                and not isabstract(m)
+                                                and issubclass(m, Parser)
+                                                and m is not Parser))
+
+        return classes[0][1]
```

Comment:
Will update to say a class is returned

Reply:
Is this the same for your other factory methods?  I don't think it matters which you pick but it would probably be good to be consistent as to whether an object or class is returned (unless there is a good reason for the difference obviously!).

Reply:
Yeah they both return the class without initialising